.pre-commit-config.yaml

```diff
@@ -0,0 +1,48 @@
+---
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v5.0.0
+    hooks:
+      - id: check-added-large-files
+      - id: check-case-conflict
+      - id: check-executables-have-shebangs
+      - id: check-shebang-scripts-are-executable
+      - id: check-symlinks
+      - id: destroyed-symlinks
+
+      - id: check-toml
+      - id: check-json
+      - id: check-yaml
+        args: [--allow-multiple-documents]
+        exclude: ^(deploy/.*|chart/templates/.*)$
+
+      - id: check-merge-conflict
+      - id: end-of-file-fixer
+      - id: mixed-line-ending
+        args: [--fix=lf]
+      - id: trailing-whitespace
+        exclude: ^(chart/.snapshots/.*)$
+
+  - repo: https://github.com/pre-commit/mirrors-prettier
+    rev: v3.1.0
+    hooks:
+      - id: prettier
+        files: \.(md|ya?ml|json)$
+        exclude: ^(CHANGELOG.md|chart/templates/.*|chart/.snapshots/.*)$
```

Comment:
I'd rather keep the snapshots in there, if pre-commit fails on the snapshots, the templates might need a fix.

Reply:
Running prettier and helm template causes some conflicts. pre-commit (prettier) wants to format the yaml. In the CI we check for a diff after running `helm template`, which always fails as the formatting is different.

Should we write a custom pre-commit hook to address this?